docs/python/formatting.md

```diff
@@ -0,0 +1,83 @@
+
+
+Alternatively, you can set it as the default formatter for all Python files by setting `"editor.defaultFormatter"` in your User `settings.json` file, under a `[python]` scope.
+
+For example, to set Black Formatter as the default formatter, add the following setting to your User `settings.json` file:
+
+```
+  "[python]": {
+    "editor.defaultFormatter": "ms-python.black-formatter"
+  }
+```
+
+### Formatting your code
+
+You can format your code by right-clicking on the editor and selecting **Format Document**, or by using the `kb(editor.action.formatDocument)` keyboard shortcut.
+
+You can also add the following setting to your User `settings.json` file to enable formatting on save for your code:
+
+```
+  "[python]": {
+    "editor.formatOnSave": true
+  }
+```
+
+### General formatting settings
+
+Each formatter extension may have its own settings, but the ones below are supported by both [autopep8](https://marketplace.visualstudio.com/items?itemName=ms-python.autopep8) and [Black Formatter](https://marketplace.visualstudio.com/items?itemName=ms-python.black-formatter):
+
+| Setting Suffix<br/> | Default value | Description |
+| --- | --- | --- |
+| args | `[]` | Arguments to be passed to the formatter. Each argument should be passed as a separate string in the array. <br> For example:<br> `black-formatter.args: ["--line-length", "100"]` |
+| importStrategy | `useBundled` | When set to `useBundled`, the extension uses the version of the tool that it ships with. When set to `fromEnvironment`, it attempts to load from your selected Python environment first, otherwise it falls back to the bundled version. |
+| path | `""` | Path to the formatter binary to be used for formatting. **Note:** Using this option may slowdown formatting. |
+| interpreter | `[]` | When set to a path to a Python executable, the extension will use that to launch the formatter server and its subprocesses. |
+| showNotifications | `off`| Controls when notifications are displayed by this extension. Supported values are `off`, `always`, `onError`, and `onWarning`. |
+
+
+### Troubleshooting formatting
+
+If formatting fails, check the following possible causes:
+
+| Problem | Solution |
+| --- | --- |
+| There are multiple formatters available for Python files | Set the default formatter by following the instructions in [the section above](#setting-a-default-formatter). |
+| Custom arguments for the formatter are incorrect. | Check that the appropriate `<formatter>.path` setting does not contain arguments, and that `<formatter>.args` contains a list of individual top-level argument elements. |
+| Format Selection fails with Black Formatter | `black` does not support formatting sections of code. To work around this limitation, you can disable format on paste and set `formatOnSave`` to format the whole file with the following settings: `"[python]": {"editor.formatOnPaste": false, "editor.formatOnSaveMode": "file"}`.|
```

Comment:
```suggestion
| Format selection fails with Black Formatter | `black` does not support formatting sections of code. To work around this limitation, you can disable format on paste and set `formatOnSave`` to format the whole file with the following settings: `"[python]": {"editor.formatOnPaste": false, "editor.formatOnSaveMode": "file"}`.|
```

Reply:
because it's a command name, I think I'll leave capitalized but will bold it for consistency